contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -719,6 +725,67 @@ plsql_TriggerRecursiveCheck(ResultRelInfo *resultRelInfo)
 	return false;
 }
 
+/**
+ * Hook function to skip rewriting VIEW with base table if the VIEW has an instead of trigger
+ * Checks if view have an INSTEAD OF trigger at statement level
+ * If it does, we don't want to treat it as auto-updatable.
+ * This function also does error checking for recursive triggers
+ * Reference - src/backend/rewrite/rewriteHandler.c view_has_instead_trigger
+ */
+static bool
+pltsql_bbfViewHasInsteadofTrigger(Relation view, CmdType event)
+{
+	TriggerDesc *trigDesc = view->trigdesc;
+	if (trigDesc && triggerInvocationSequence)
+	{
+		int i;
+		for (i = 0; i < trigDesc->numtriggers; i++)
+		{
+			Trigger *trigger = &trigDesc->triggers[i];
+			Oid current_tgoid = trigger->tgoid;
+			Oid prev_tgoid = InvalidOid;
+			prev_tgoid = lfirst_oid(list_tail(triggerInvocationSequence));
+			if (list_length(triggerInvocationSequence) > 32)
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_SYNTAX_ERROR),
+						 errmsg("Maximum stored procedure, function, trigger, or view nesting level exceeded (limit 32)")));
+			}
+			else if (prev_tgoid == current_tgoid)
+			{
+				return false; /** A trigger called recursively by itself*/
+			}
+			else if (list_member_oid(triggerInvocationSequence, current_tgoid))
+			{
+				/** A trigger called recursively by another trigger */
+				ereport(ERROR,
+						(errcode(ERRCODE_SYNTAX_ERROR),
+						 errmsg("Maximum stored procedure, function, trigger, or view nesting level exceeded (limit 32)")));
```

Comment:
This can be combined with previous if check. And this is not a syntax error, ERRCODE_PROGRAM_LIMIT_EXCEEDED may be a better error code.
The comment is also a bit confusing here - I suggest change it into something like "loop trigger call".